src/dialect/sqlite.rs

```diff
@@ -0,0 +1,39 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::dialect::Dialect;
+
+/// For resilience when confronted with historical SQL statements, SQLite will sometimes bend the quoting rules above:
+///     * If a keyword in single quotes (ex: 'key' or 'glob') is used in a context where an identifier is allowed but where a string literal is not allowed, then the token is understood to be an identifier instead of a string literal.
+///     * If a keyword in double quotes (ex: "key" or "glob") is used in a context where it cannot be resolved to an identifier but where a string literal is allowed, then the token is understood to be a string literal instead of an identifier.
+#[derive(Debug)]
+pub struct SQLiteDialect {}
+
+impl Dialect for SQLiteDialect {
+    // see https://www.sqlite.org/lang_keywords.html
+    fn is_delimited_identifier_start(&self, ch: char) -> bool {
+        ch == '`' || ch == '\'' || ch == '"' || ch == '['
+    }
```

Comment:
The issue I was pointing to is that this comment doesn't help understanding why `is_delimited_identifier_start` is implemented the way it is. The copy-pasta you've added above doesn't help with that.

Reply:
From the document, we know that SQLite supports single quotes, double quotes, square brackets and grave accents. It is applicable not only to keywords but also to other identifiers. SQLite will sometimes bend the quoting rules for  historical SQL statements, but after that it says
>  Programmers are cautioned not to use the two exceptions described in the previous bullets. We emphasize that they exist only so that old and ill-formed SQL statements will run correctly. Future versions of SQLite might raise errors instead of accepting the malformed statements covered by the exceptions above.

> SQLite adds new keywords from time to time when it takes on new features. So to prevent your code from being broken by future enhancements, you should normally quote any identifier that is an English language word, even if you do not have to.

Reply:
The document says that SQLite uses `""`, `[]`, and <code>&#96;&#96;</code> for identifiers, but sometimes, depending on the context, treats `'string literals'` as identifiers too, and `"double-quoted identifiers"` as string literals. You make all four of `""`, `[]`, <code>&#96;&#96;</code>, `''` unconditionally be parsed as identifiers, which might be fine, but can't be documented by simply linking to the SQLite's page. Someone will come in later and ask why the string literals are not parsed as such, and won't be able to answer to that.

Reply:
Thanks for pointing out that, I think now I will just use ` `` ` and `[]` for identifiers. In most cases, single quotes and double quotes should be passed as strings, it's a little complex to implement parse that depending on the context, maybe I can implement it in the next PR. So I will comment that
 > Use ```` and `[]` for identifiers 
 > TODO: Depending on the context, treats 'string literals' as identifiers too, and "double-quoted identifiers" as string literals.